### DISCUSSION TITLE
How do I add additional hostRules in specific repositories

### DISCUSSION THREAD

**Discussion by michaelswells** [renovatebot/renovate#32886] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, renovate tag is 38.15.1

### Please tell us more about your question or problem

We have a renovate repo `renovate-bot` in the `custom-apps/devops/` group folder that has the following: files
`default.json`
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:recommended",
    ":semanticCommits",
    "group:recommended"
  ]
}
```
and
`config.js`
```javascript
module.exports = {
    endpoint: 'https://gitlab.us.*.com/api/v4/',
    token: process.env.DEPLOYER_TOKEN,
    platform: 'gitlab',
    enabledManagers: ['nuget'],
    prHourlyLimit: 0,
    rebaseWhen: "behind-base-branch",
    dependencyDashboard: true,
    autodiscover: true,
    autodiscoverFilter: [
      "custom-apps/devops/*",
      "custom-apps/engineering/apps/**/*",
      "custom-apps/group1/*",
      "custom-apps/group2/*"
    ],
    onboardingConfig: {
      extends: [
        "local>custom-apps/devops/renovate-bot",
        "local>custom-apps/devops/renovate-bot:shrapnel"
      ],
      hostRules: [
        {
          matchHost: "nexus.global.*.com",
          username: process.env.EFOSS_USER,
          password: process.env.EFOSS_TOKEN
        }
      ]
    },
    hostRules: [
      {
        matchHost: "nexus.global.*.com",
        username: process.env.EFOSS_USER,
        password: process.env.EFOSS_TOKEN
      }
    ]
  };
  ```
My team's repositories are all located under `custom-apps\engineering\apps` group folder and we would need to add a second hostRule with different credentials, e.g.,
```
      {
        matchHost: "nexus-ent.us.*.com",
        username: process.env.NEXUS_US_USERNAME,
        password: process.env.NEXUS_US_TOKEN
      }
```

My current `renovate.json` file located in one of my repos contains:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>custom-apps/devops/renovate-bot",
    "local>custom-apps/devops/renovate-bot:shrapnel"
  ]
}
```

I cannot find (or understand) the documentation on how I can achieve this. We have around 20 repositories under `engineering` and I would need to  make the additional hostRule available to each project.

How would I accomplish this? local `renovate.json` file? local `config.js` file? any other solution? I could have a new file to the `renovate-bot` project added if that would help.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Do you mean that there's around 20 repos which all need to share the same host rules credential?

First of all, is there a problem if you add it to the global config? It does not seem to conflict with your first host rule.
